client/test/IVY-58.spec.js

```diff
@@ -1,58 +1,80 @@
-            driver.wait(until.elementIsSelected(password));
-            password.sendKeys("password");
+            let username = await driver.findElement(By.id("email-input-field"));
+            let password = await driver.findElement(
+                By.id("filled-password-input")
+            );
+            let login = await driver.findElement(By.id("login-button"));
+
+            await username.sendKeys("admin@unb.ca");
+            await password.sendKeys("password");
             login.click();
-            
-            let loggedin = driver.findElement(By.xpath("./div[@class='header-name']"));
+            await driver.wait(
+                until.elementLocated(By.className("header-name"))
+            );
+            // Wait for the App to render
+            let token = await driver.executeScript(
+                'return sessionStorage.getItem("token")'
+            );
+            assert.notEqual(token, undefined);
+            // let timer = await driver.wait(until.elementLocated(By.xpath("//a[@href='/teacherAbsences']")), 10000);
+
+            // URL does not change during navigation with our current implementation of header
+            // This may change in the future
+            // let url = await driver.getCurrentUrl();
+            // assert.equal("http://localhost:3000/teacherAbsences", url);
+            // await driver.manage().setTimeouts({ implicit: 500 });
+            /**/
+        });
+
+        it('Navigate to Absences', async function() {
+            await driver.get("http://localhost:3000");
+            await driver.manage().setTimeouts({ implicit: 500 });
 
-            assert.equal('header-name', loggedin);
-            
-            /*let timer = await driver.wait(until.elementLocated(By.xpath("//a[@href='/teacherAbsences']")), 10000);
             let title = await driver.getTitle();
 
-            assert.equal('React App', title);
+            assert.equal("React App", title);
 
-            await driver.wait(until.elementLocated(By.xpath("//a[@href='/teacherAbsences']")));
-            let found = driver.findElement(By.xpath("//a[@href='/teacherAbsences']"));
-            driver.wait(until.elementIsSelected(found));
+            await driver.wait(
+                until.elementLocated(By.linkText("Teacher Absences"))
+            );
+            let found = await driver.findElement(
+                By.linkText("Teacher Absences")
+            );
+            // await driver.wait(until.elementIsSelected(found));
             found.click();
-            let waiting = await driver.wait(until.elementLocated(By.xpath("//div[@class='teacherAbsenceForm']")), 10000);
-            let url = await driver.getCurrentUrl();
-            assert.equal('http://localhost:3000/teacherAbsences', url);
-            await driver.manage().setTimeouts({ implicit: 500 });*/
-        });
+            const iframe = await driver.findElement(By.xpath("//iframe[@name='frame']"));
```

Comment:
This will fail once the dashboard has been updated (https://github.com/ivysaur-unb/on-call-coordinator/pull/29). But I think it will just be this line that will need to update

Reply:
Thanks, we should fix this with whichever PR gets merged last. I'm thinking we just have to delete the 2 lines for switching frames